BundleTransport/app/src/main/java/com/ddd/wifidirect/WifiDirectBroadcastReceiver.java

```diff
@@ -59,10 +59,9 @@ public void onReceive(Context context, Intent intent) {
                 Log.d(MainActivity.TAG, "P2P state changed Wifi Direct is enabled - " + state);
                 // auto start gRPC server when app is opened
                 Log.d(MainActivity.TAG, "Starting server, at least one device is connected to group");
-                Data data = new Data.Builder().putInt("PORT", 1778).build();
                 PeriodicWorkRequest request =
                         new PeriodicWorkRequest.Builder(RpcServerWorker.class, 15, TimeUnit.MINUTES, 15,
-                                                        TimeUnit.MINUTES).setInputData(data).build();
```

Comment:
you might check with @JAReddy on this. i'm not sure we need this timer.

Reply:
Since we are checking the status of the server before actually starting it, so this should Ideally avoid duplicate startups. Ideally the server should not go down without any interference but I think this should work as health check service and should not have any error logs like before. Please verify if the server logs are clean for at least 15+ minutes( i.e periodicity of the request)